src/containers/ProjectsList.js

```diff
@@ -1,8 +1,208 @@
-import React, { Component } from 'react'
-import { Header } from 'semantic-ui-react'
+import React, { Component, Fragment } from 'react'
+import { Header, Card, Grid } from 'semantic-ui-react'
+import { connect } from 'react-redux'
+import { fetchProjects } from '../actions/getProjectsAction'
+import { Link } from 'react-router-dom'
+import Select from 'react-select'
+import Project from '../components/Project'
+import Paginate from '../components/Paginate'
+import PaginationLinks from '../components/PaginationLinks'
+import '../assets/ProjectsList.css'
 
 export class ProjectsList extends Component {
+  state = {
+    firstPage: true,
+    lastPage: true,
+    pageCount: null,
+    projectsList: [],
+    projects: {},
+    selectedPage: 1,
+    filteredProjectsList: null,
+    filteredProjects: {},
+    totalProjects: null,
+    selectedLanguage: null,
+    languages: []
+  };
+
+  componentDidMount () {
+    if (!this.props.projects.length) {
+      this.props.fetchProjects()
```

Comment:
I believe what happens is that if the `fetchProjects` is not successful it will just show the spinner... you can go to the /users page right now while the server is off and see this use case

Reply:
I didnt try it yet. but I am not sure the intended behavior is for the spinner to be there forever. That tells me that it is trying and even if the server gets back up we will not get the projects.

This PR is already humongus, so maybe we can roll with this and I will create a bug for it